### DISCUSSION TITLE
Wrong gt while using browse_dataset.py

### DISCUSSION THREAD

**Discussion by hanssssssss** [open-mmlab/mmdetection3d#2074] [Q&A] [answered]
Hi, I tried to visualize the waymo or kitti dataset on the latest version 1.1. I found that the gt bounding box shifts up on every image on both datasets.Moreover, I found that the 3dbbox did not transform with the image when using DA.
![3](https://user-images.githubusercontent.com/57829884/204960955-7f001bb3-760d-43e7-9841-ea59496a208c.jpg)

![23](https://user-images.githubusercontent.com/57829884/204960898-e64ee0c8-c369-4310-97e2-d18d7ed58e8e.jpg)

**Comment by ZCMax** [comment, marked as answer]
Are you using the latest dev-1.x branch or 1.1 branch?

**Reply by hanssssssss** [reply]
Hi, I used the 1.1 branch. Btw, I wonder if the label while transform correctly when I used DA during training.

**Reply by ZCMax** [reply]
Can you try to switch to the latest dev-1.x branch? The code supports label transform while DA during training.

**Reply by hanssssssss** [reply]
Hi, Thanks for the help. I will try it. Do you mean all the branches(master, 1.1, dev-1.x) support the label transform while using DA during training? Thanks again.

**Reply by ZCMax** [reply]
Yes, `browse_dataset.py` script supports visualizing the final transformed results during training.

**Reply by hanssssssss** [reply]
Hi, I used the 1.x branch, it seems okay now. But I found that when using the `affineresize` in DA, the visualisation result will sitll shift. If I remove the operation, it becomes okay. Is it a bug?
![10](https://user-images.githubusercontent.com/57829884/204985499-c6c0a448-96db-4900-9172-385c82eb5d75.jpg)
![11](https://user-images.githubusercontent.com/57829884/204985508-6fedbc98-6d24-4e0a-b054-4790e4fcd76d.jpg)

**Reply by ZCMax** [reply]
It's not a bug, since in affineresize DA, 3d bboxes of images which are affined will not be used to calculate loss.

**Reply by hanssssssss** [reply]
Hi, do you mean when I used the affineresize DA during training, the 3d bboxes labels are of no use for the model, so they will not be used to calculate loss? Thanks again.

**Reply by ZCMax** [reply]
For the image applied affineresize, only 2d losses will be calculated,  3d losses associating 3D bboxes won't be calculated, which means at this time the 3d bboxes labels are of no use for the model. But for the image not applied affinereisze, 2d and 3d losses will both be calculated.

**Reply by hanssssssss** [reply]
Hi! Thanks for the help! I noticed that the smoke using affineresize during training, since it is for mond3d task, will the affineresize help to converge? Besides, I tried to remove the affineresize in the smoke pipeline, but something went wrong.

**Reply by ZCMax** [reply]
Yes, affineresize will be useful for `center_2d` localization, and now, smoke is bound with this pipeline, but you can modify the code in   `SMOKEMono3DHead` to support removing affineresize during training.

**Reply by hanssssssss** [reply]
Got it! I have one more question(last one :) ), the affineresize in the pieline is applied all the time or has a certain probability to be applied? If it is used all the time, the 3d labels will not be used during training as you said. Many thanks.

**Reply by ZCMax** [reply]
a certain probability, affineresize is bound with RandomShiftScale pipeline, they are used together, more details can be found in our code.

**Reply by hanssssssss** [reply]
got it. Thanks for the patience
